### PR TITLE
API 7558 PractitionerRole Datamart Model

### PR DESCRIPTION
**Pull request by that-guy-is-it**

### CODE REVIEW

**Comment by that-guy-is-it** [comment]
@nmoisan  @jtaylor-va  @haidong I realize I forgot to add a ping on the description, so I am adding it here as a comment.

**Review by nmoisan — APPROVED** [review]
LGTM

**Review by bschofield-va — DISMISSED** [review]

**Review by haidong — APPROVED** [review]

File: db/migration/V17_0__create_practitioner_role.sql [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+(
+  [CDWIdNumber] [int] not null,
+  [CDWIdResourceCode] [char](1) not null,
+  [Sta3n] [smallint] not null,
+  [Specialty] [varchar](100) null,
+  [PractitionerGivenName] [varchar](50) null,
+  [PractitionerFamilyName] [varchar](50) null,
+  [PractitionerNPI] [varchar](50) null,
+  --JSON Payload
+  [PractitionerRole] [varchar](MAX),
```
Comment (jblefkove-va) [inline db/migration/V17_0__create_practitioner_role.sql:11]: should this also be `not null`?

File: db/migration/V17_0__create_practitioner_role.sql [hunk 2]
```diff
@@ -0,0 +1,13 @@
+Create table [App].[PractitionerRole]
+(
+  [CDWIdNumber] [int] not null,
+  [CDWIdResourceCode] [char](1) not null,
+  [Sta3n] [smallint] not null,
+  [Specialty] [varchar](100) null,
```
Comment (jblefkove-va) [inline db/migration/V17_0__create_practitioner_role.sql:6]: @nmoisan any concerns about 100 characters supporting multiple specialties?

File: db/migration/V17_0__create_practitioner_role.sql [hunk 3]
```diff
@@ -0,0 +1,13 @@
+Create table [App].[PractitionerRole]
+(
+  [CDWIdNumber] [int] not null,
+  [CDWIdResourceCode] [char](1) not null,
+  [Sta3n] [smallint] not null,
```
Comment (jblefkove-va) [inline db/migration/V17_0__create_practitioner_role.sql:5]: Do we need `sta3n` column? I don't think we search on this one.

Comment (nmoisan) [reply db/migration/V17_0__create_practitioner_role.sql:5]: I doubt it, was just matching existing filters within `Practitioner` when i wrote up the schema last week.

File: db/migration/V17_0__create_practitioner_role.sql (hunk 2 above)
Comment (nmoisan) [reply db/migration/V17_0__create_practitioner_role.sql:6]: The max value within the table is 22 characters. perhaps we can expand it to 255 here just to be on the safe side? (if we go the route of aggregating in the db).

Comment (jblefkove-va) [reply db/migration/V17_0__create_practitioner_role.sql:6]: I'm good with 255

File: db/migration/V17_0__create_practitioner_role.sql (hunk 3 above)
Comment (jblefkove-va) [reply db/migration/V17_0__create_practitioner_role.sql:5]: Let's omit this column for now then

**Review by jblefkove-va — CHANGES_REQUESTED** [review]
see inline

**Review by jblefkove-va — APPROVED** [review]

**Pull request merged**